cunumeric/_ufunc/ufunc.py

```diff
@@ -571,19 +571,20 @@ def _find_common_type(
         )
         use_min_scalar = not (all_scalars or all_arrays or lossy_conversion)
```

Comment:
```suggestion
        use_min_scalar = not (all_scalars or all_arrays)
```
Aha, I missed that the old code had this oddity; probably because `np.find_common_type` was weird...  It is not always correct this way and `result_type` applies the correct logic.
(The only reason this ever worked must be because `find_common_dtype` didn't even match how numpy does promotion, apparently...)

Anyway, removing this, the eager tests seem to passed locally for me.  (The `use_min_scalar` logic could be removed, always taking the `True` path is correct with `result_type`)